network/net.go

```diff
@@ -505,29 +549,47 @@ func (sc *SecureTCPConn) ServerIdentity() *ServerIdentity {
 // exchangeServerIdentity is made to exchange the ServerIdentity between the two parties.
 // when a connection request is made during listening
 func (sc *SecureTCPConn) exchangeServerIdentity() error {
-	ourEnt := sc.SecureTCPHost.entity
+	ourEnt := sc.SecureTCPHost.serverIdentity
 	if ourEnt == nil {
 		ourEnt = NewServerIdentity(config.NewKeyPair(Suite).Public, "")
 	}
 	// Send our ServerIdentity to the remote endpoint
-	log.Lvl4("Sending our identity", ourEnt.ID, "to",
+	log.Lvlf4("Sending our identity %x to %s", ourEnt.ID,
 		sc.TCPConn.conn.RemoteAddr().String())
 	if err := sc.TCPConn.Send(context.TODO(), ourEnt); err != nil {
-		return fmt.Errorf("Error while sending indentity during negotiation:%s", err)
+		return fmt.Errorf("Error while sending indentity during negotiation: %s", err)
+	}
+
+	// Try for 1 second to receive the other entity
+	var nm Packet
+	var err error
+	for i := 0; i < 10; i++ {
+		// Receive the other ServerIdentity
+		nm, err = sc.TCPConn.Receive(context.TODO())
+		switch {
+		case err == nil:
+			if i > 0 {
+				log.Warn(sc, "Got a packet after a failure")
```

Comment:
Yes, I had this case from time to time, but not lately. So I wanted to check if it still happens. I can also just take it away. 

The handling here happens if the connection is opened and the listener tries to receive bytes before the identity has been sent. Which seems to be plausible to me, but I didn't see it anymore (I did so earlier). Should perhaps again be a test which does this.